boot/freeldr/freeldr/pcat.cmake

```diff
@@ -110,6 +110,9 @@ if(ARCH STREQUAL "i386")
     endif()
 
 elseif(ARCH STREQUAL "amd64")
+    list(APPEND PCATLDR_BASE_ASM_SOURCE
+        arch/i386/multiboot.S)
```

Comment:
@tkreuzer adding a file from i386 to the amd64-build... Is that a legit thing to do?

Reply:
Timo has been commented on this in the chat: ![image](https://github.com/user-attachments/assets/82960055-ab34-49ba-a92c-93be237994df)

Reply:
I added the i386 multiboot code because the code is compatible and avoid duplication.

Reply:
Well, but you have to make the x64 bots compile at least. Currently they do fail with your PR. So porting that ASM magic seems to be the only legit solution to me.

Reply:
Okay now everything is compiling fine. now the MSVC builds is crashing.